packages/bitcoin/test/BitcoinWallet.test.ts

```diff
@@ -0,0 +1,316 @@
+    expect(fees.fast.feeRate).toBe(0.000_025);
+    expect(fees.standard.feeRate).toBe(0.000_015);
+    expect(fees.slow.feeRate).toBe(0.000_01);
+
+    wallet.shutdown();
+  });
+
+  it('populates utxos$ when new UTXOs are received', async () => {
+    const utxos: UTxO[] = [{ txId: 'abc', index: 0, address: 'addr1', satoshis: BigInt(1000) }];
+
+    provider.getUTxOs.mockResolvedValue(utxos);
+
+    const poll$ = of(false);
+    const wallet = new BitcoinWallet(provider, 30_000, 20, walletInfo, Network.Testnet, poll$, logger);
+
+    expect(wallet.utxos$.value).toEqual([]);
+
+    await (wallet as any).updateUtxos();
+
+    expect(wallet.utxos$.value).toEqual(utxos);
+
+    wallet.shutdown();
+  });
+
+  it('updates utxos$ and balance$ when the UTXO set changes', async () => {
+    const oldUtxos: UTxO[] = [{ txId: 'abc', index: 0, address: 'addr1', satoshis: BigInt(1000) }];
+
+    const newUtxos: UTxO[] = [
+      { txId: 'def', index: 1, address: 'addr2', satoshis: BigInt(2000) },
+      { txId: 'ghi', index: 2, address: 'addr3', satoshis: BigInt(3000) }
+    ];
+
+    provider.getUTxOs.mockResolvedValue(newUtxos);
+
+    const poll$ = of(false);
+    const wallet = new BitcoinWallet(provider, 30_000, 20, walletInfo, Network.Testnet, poll$, logger);
+
+    wallet.utxos$.next(oldUtxos);
+    wallet.balance$.next(BigInt(1000));
+
+    await (wallet as any).updateUtxos();
+
+    expect(wallet.utxos$.value).toEqual(newUtxos);
+    expect(wallet.balance$.value).toEqual(BigInt(5000));
+
+    wallet.shutdown();
+  });
+
+  it('updates transaction history when new transactions are detected', async () => {
+    const poll$ = of(false);
+    const wallet = new BitcoinWallet(provider, 30_000, 20, walletInfo, Network.Testnet, poll$, logger);
+
+    const newTxs: TransactionHistoryEntry[] = [
+      {
+        transactionHash: 'tx1',
+        inputs: [],
+        outputs: [],
+        confirmations: 1,
+        blockHeight: 100,
+        timestamp: 12_345_678
+      } as unknown as TransactionHistoryEntry
+    ];
+
+    provider.getTransactions.mockResolvedValue({ transactions: newTxs });
+
+    const changed = await (wallet as any).updateTransactions();
+
+    expect(changed).toBe(true);
+    expect((wallet as any).transactionHistory).toEqual(newTxs);
+
+    const emitted = await firstValueFrom(wallet.transactionHistory$);
+    expect(emitted).toEqual(newTxs);
+
+    wallet.shutdown();
+  });
+
+  it('does not update transaction history when there are no changes', async () => {
+    const poll$ = of(false);
+    const wallet = new BitcoinWallet(provider, 30_000, 20, walletInfo, Network.Testnet, poll$, logger);
+
+    const existingTxs: TransactionHistoryEntry[] = [
+      {
+        transactionHash: 'tx1',
+        inputs: [],
+        outputs: [],
+        confirmations: 1,
+        blockHeight: 100,
+        timestamp: 12_345_678
+      } as unknown as TransactionHistoryEntry
+    ];
+
+    wallet.transactionHistory = existingTxs;
```

Comment:
To resolve the error here and match the approach in this test file
```suggestion
    (wallet as any).transactionHistory = existingTxs;
```

Reply:
done here https://github.com/input-output-hk/lace/pull/1855/commits/d501a183e05621c6cf939486b60caf9e2ad2f1a0